core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java

```diff
@@ -634,6 +496,14 @@ private void makeFactoryDump(RudderMessage message, boolean fromHistory) {
         }
     }
 
+    void flushSync() {
+//        synchronized (this){
```

Comment:
What is the current behaviour when multiple clients are calling flushSync and the previous one is still in progress?

Reply:
We have used synchronised blocks to synchronize database access and server upload. We also have a test case around this.

Reply:
I think what we discussed in the call is:

Assume there are multiple callers (c1, c2, c3)

At t1 -> c1.flush()
At t2 -> c2.flush()
At t3 -> c3.flush()

At t4 -> c1.flush() finishes
Will the pending flushes (c2 & c3) be merged?


Right now, we have a FLUSH_LOCK, which I believe is synchronizing writing to the output stream. Not sure how that helps. 
We should be able to merge and make a single call/fewer calls.